### PR TITLE
Refactor event_sample_rate to analytics

### PR DESCRIPTION
**Pull request by delner**
This pull request:

 - Renames `Datadog::Contrib::Sampling` to `Datadog::Contrib::Analytics`
 - Changes `event_sample_rate` for current integrations to `analytics_sample_rate` and `analytics_enabled`. (Breaking change)
 - Adds environment variables for each of these configuration options.

It does not implement the global setting, which will be done in another pull request.

Serves a prerequisite for #697.

### CODE REVIEW

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -518,7 +518,8 @@ Where `options` is an optional `Hash` that accepts the following parameters:
 
 | Key | Description | Default |
 | --- | ----------- | ------- |
-| `event_sample_rate` | Rate which spans should be sampled for search and analytics. | `nil` |
+| `analytics_enabled` | Enable analytics for spans produced by this integration. `true` for on, `nil` to defer to global setting, `false` for off. | `nil` |
+| `analytics_sample_rate` | Rate which spans should be sampled for analytics. | `1.0` |
```
[the remark this answers is not in the document]
Comment (delner) [reply docs/GettingStarted.md]: Okay, so strike all references of `analytics_enabled` and `analytics_sample_rate` from the documentation?

File: spec/ddtrace/contrib/analytics_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+require 'spec_helper'
+
+require 'ddtrace/contrib/analytics'
+
+RSpec.describe Contrib::Analytics.set_sample_rate do
```
[the remark this answers is not in the document]
Comment (delner) [reply spec/ddtrace/contrib/analytics_spec.rb]: "?" indeed...  let me check this real quick...

File: docs/GettingStarted.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (brettlangdon) [reply docs/GettingStarted.md]: Just `analytics_sample_rate`, we want to keep `analytics_enabled` documented.

Sorry, should have clarified here.

File: spec/ddtrace/contrib/analytics_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (delner) [reply spec/ddtrace/contrib/analytics_spec.rb]: This test was actually not running as it should have. Fixed now. Good catch.

**Pull request merged**